com.unity.shadergraph/Editor/Data/Nodes/Input/PropertyNode.cs

```diff
@@ -225,6 +225,11 @@ public void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode generationMo
                         sb.AppendLine($"Gradient {GetVariableNameForSlot(OutputSlotId)} = {property.GetHLSLVariableName(isGeneratingSubgraph)};");
                     break;
             }
+
+            if (property.isConnectionTestable)
+            {
+                sb.AppendLine("bool {0} = {1};", GetConnectionStateVariableNameForSlot(OutputSlotId), (generationMode == GenerationMode.Preview || !isGeneratingSubgraph) ? (IsSlotConnected(OutputSlotId) ? "true" : "false") : property.GetConnectionStateHLSLVariableName());
```

Comment:
Right, so in preview mode code generation, the shader gets built slightly differently, since it doesn't resolve the property values the same way, and code gets inlined where possible. The logic in there is specifically to handle whether or not the value is inlined directly, or needs to pull from the associated variable. I'm not sure we have a better way to abstract that. Check out something like ColorNode for a more complex example of something along these lines.

Reply:
Done.